ui/src/message_popup/author/new_question_page.jsx

```diff
@@ -0,0 +1,50 @@
+/* eslint no-console: "off" */
+import React from 'react';
+import _ from 'lodash';
+
+import QuestionPage from './question_page.jsx';
+import * as Api from '../../helpers/api.js';
+
+
+export default React.createClass({
+  displayName: 'NewQuestionPage',
+
+  propTypes: {
+    loaded: React.PropTypes.bool,
+    allQuestions: React.PropTypes.object,
+    onReloadQuestions: React.PropTypes.func
+  },
+
+  componentDidMount(){
+    this.props.onReloadQuestions();
+  },
+
+  getNewID(){
+    const {allQuestions} = this.props;
+    const allQuestionsFlat = allQuestions.currentQuestions.concat(allQuestions.archivedQuestions);
+    const mostRecentQuestion = _.maxBy(allQuestionsFlat, question => question.id);
+    const largestID = mostRecentQuestion !== undefined ? mostRecentQuestion.id : 0;
+    return largestID + 1;
+  },
+
+  onCreateQuestion(newQuestion){
+    const {currentQuestions, archivedQuestions} = this.props.allQuestions;
+    const question = {id: this.getNewID(), ...newQuestion};
+    const newCurrentQuestions = currentQuestions.concat(question);
+    console.log('Creating new question...');
+    console.log(question);
+    Api.saveQuestions({currentQuestions: newCurrentQuestions, archivedQuestions});
```

Comment:
Ahh, I think I see more why you were talking about mixing the "update/edit/create" capability with the loader capability, since this needs a way to update the state of questions as well as making the server call.

Let's try this approach: if we move the loader component into each page (out of `app`), and add another prop to `QuestionLoader` for updating the questions (eg., `onQuestionsUpdated`), we can the move the server call into the loader as I think you originally suggested.